improver/psychrometric_calculations/cloud_top_temperature.py

```diff
@@ -112,23 +114,45 @@ def _make_cct_cube(self, data: ndarray) -> Cube:
         )
         return cube
 
-    def process(self, t_at_ccl: Cube, p_at_ccl: Cube, temperature: Cube) -> Cube:
+    def process(self, *cubes: Union[Cube, CubeList]) -> Cube:
         """
 
         Args:
-            t_at_ccl:
-                temperature at cloud condensation level
-            p_at_ccl:
-                pressure at cloud condensation level
-            temperature:
-                temperature on pressure levels
+            cubes:
+                Cubes, of 'temperature at cloud condensation level',
+                'pressure at cloud condensation level' and 'temperature on pressure levels'.
 
         Returns:
             Cube of cloud top temperature
         """
-        self.t_at_ccl = t_at_ccl
-        self.p_at_ccl = p_at_ccl
-        self.temperature = temperature
+        cubes = as_cubelist(*cubes)
+        self.t_at_ccl = cubes.extract(
+            [
+                "air_temperature_at_condensation_level",
+                "air_temperature_at_cloud_condensation_level",
+            ]
+        )
+        self.p_at_ccl = cubes.extract(
+            [
+                "air_pressure_at_condensation_level",
+                "air_pressure_at_cloud_condensation_level",
+            ]
+        )
+        self.temperature = cubes.extract(
+            ["temperature_on_pressure_levels", "air_temperature"]
+        )
```

Comment:
These alternate names were provided in order to ensure that the tests would still succeed.
Perhaps the tests don't have the correct names and need updating instead of this flexible extract?

Alternative, if we need to be more relaxed about naming here, we can do something like I see you have done elsewhere.  That is, extract all the cube names, then match against the names containing:

- "air_temperature"
- "air_pressure"
- remaining cube containing "temperature"

I'm all 👂's